include/rmm/mr/device/device_memory_resource.hpp

```diff
@@ -117,6 +118,7 @@ class device_memory_resource {
    */
   void* allocate(std::size_t bytes, cuda_stream_view stream = cuda_stream_view{})
   {
+    RMM_FUNC_RANGE();
```

Comment:
*question*: Do we want the same set of annotations for the `host_memory_resource` as well?

Reply:
Yes. I added them. Note that the new `pinned_host_memory_resource` does not derive from `host_memory_resource` so I had to add ranges to it explicitly. This is the way it will have to be done for all MRs that just implement the concepts in the future, unfortunately.